### DISCUSSION TITLE
Comment contribuer à publicodes ?

### DISCUSSION THREAD

**Discussion by laem** [publicodes/publicodes#93] [Q&A] [answered]
@mquandalle @johangirod on peut voir qu'une PR sur betagouv/mon-entreprise est en train d'améliorer publicodes, au moins sa partie `publicodes-react`, [celle-ci](https://github.com/betagouv/mon-entreprise/pull/1800).

Un potentiel nouveau contributeur se demandera sûrement où doit se passer la contribution : peut-il faire des PR sur ce dépôt ? Doit-il forker mon-entreprise ? 

Je pense que la page CONTRIBUTING devrait clarifier ça :)  

https://github.com/betagouv/publicodes/blob/master/CONTRIBUTING.md

**Comment by mquandalle** [comment]
Dans l'idée le développement publicodes se passe dans le dépôt publicodes.

On garde un fork sur mon-entreprise, car il nous arrive encore des faire des modifications du moteur dans le cadre de nouveaux développements spécifiques à mon-entreprise. On rapatrie ensuite les changements avec une PR dans ce dépôt, la dernière en date étant  #91. Côté mon-entreprise la chose est documentée dans notre Contributing.md https://github.com/betagouv/mon-entreprise/blob/master/CONTRIBUTING.md#modifier-publicodes

À noter que le site https://publi.codes est encore déployé depuis notre action sur le dépôt mon-entreprise. Il faudrait rapatrier ça ici.

**Reply by mquandalle** [reply]
On utilise `git subtree` mais à l'usage ce n'est finalement peut-être pas si bien. Peut-être que `git submodule` comme initialement envisagé serait meilleur. Ça nous obligerai à systématiquement séparer les commits `publicodes` de `mon-entreprise` et d'ouvrir une branche des deux côtés.

Pour des modifications comme https://github.com/betagouv/mon-entreprise/pull/1800 ça complique un peu le travail car il faut ouvrir une PR de chaque côté puis mettre à jour le sous module qu'on utilise, puis merger les deux PR.

**Comment by mquandalle** [comment, marked as answer]
Mon-entreprise utilise désormais directement les paquets `publicodes` et `publicodes-react` publiés sur npm 🎉. Tout le développement est donc enfin rappatrié dans ce dépôt.

**Reply by laem** [reply]
Génial :)

**Comment by laem** [comment]
@mquandalle : autre question. 

Pourrait-on faire un guide simple pour savoir comment gérer en tant que contributeur dev les multiples paquets qui sont ici ? 

Je veux résoudre un bug dans publicodes-react. Comment je fais mon `yarn start` pour voir les changements dans la doc docausorus par exemple ?
